dependencies.yaml

```diff
@@ -889,6 +901,12 @@ dependencies:
               - pyarrow==14.0.1
           - matrix:
             packages:
+  test_python_cudf_polars:
+    common:
+      - output_types: [conda, requirements, pyproject]
+        packages:
+          - *numpy
+          - rapids-dask-dependency==24.12.*,>=0.0.0a0
```

Comment:
This is not just a test dependency. It’s experimental but seems required for import (not just in tests)? Should it be an “extra”?

Reply:
I think that would make sense, @wence- WDYT?

Reply:
Sure. Let's make it an `experimental` extra.

Note that it's not _required_ for import, because we deliberately do not import the experimental submodule in cudf-polars proper.

Reply:
I don't know how to specify an optional `extra` for the `py_run_cudf_polars` target. If that is indeed the one we need.

Reply:
We could also just look at this in a later PR?

Reply:
Sometimes we include all feature-related extras in conda dependencies, as if they were hard requirements. We never do this for test dependencies though. It just depends on what is desired: getting an import error at runtime with a nice message and installing a package to fix it, or having the package ahead of time and paying for the extra disk space and environment solve complexity.

Reply:
As for my recommendation — until this feature is more usable / less experimental, perhaps leave it off of the conda dependencies?

Reply:
Let's leave it off for now.